tests/run.mjs

```diff
@@ -108,8 +107,8 @@ async function test() {
                     }),
                 { once: true }
             );
-            const event = new Event("start-test");
-            window.dispatchEvent(event);
+            window.runnerReady = true;
```

Comment:
I feel like it would be more self-contained to set this property inside event listener for "start-test", in index.html

Reply:
This is a scenario, where we need global variables. 
The runner shares the same window object, which is the only scope I can add a variable in the runner, that the html page "can read". If the custom event fires, before the listener was added, the html page can read its window flags and decide if it can run the tests.

Reply:
I meant that you could set the global in the `start-test` event listener above:

```js
window.addEventListener("start-test", () => {
  window.runnerReady = true;
  checkIn();
});
```

Then it's more self-contained because the globals are updated only in one file.

Reply:
What happens if the runner dispatched "start-test", before the html page can add its listener for the event?

Reply:
Ah I see what you mean. I'm not very familiar with Selenium but my understanding is that `await driver.get(...)` should be resolved only when the page is fully loaded (by default). See in https://www.selenium.dev/documentation/webdriver/drivers/options/#pageloadstrategy.

Maybe simply removing `type="module"` on the script (ensuring that it would run synchronously) would fix the issue you see, without any other change?


Otherwise I'm thinking of another way of solving this without using any variable, by adding this here:
```js
await driver.executeAsyncScript(function (callback) {
  if (document.readyState === "complete") {
    callback();
    return;
  }
  document.addEventListener("readystatechange", function onReadyStateChange() {
    if (document.readyState === "complete") {
      document.removeEventListener("readystatechange", onReadyStateChange);
      callback();
    }
  });
});
```

Reply:
I tried both and neither worked consistently

Reply:
@julienw - I refactored it a bit and using now the same solution @camillobruni did for JetStream. 
This seems to work consistently for me in CI and is a bit cleaner than before.